src/components/inputs/time-input/time-input.spec.js

```diff
@@ -1,33 +1,152 @@
 import React from 'react';
-import { shallow } from 'enzyme';
-import isTouchDevice from 'is-touch-device';
-import { TimeInput } from './time-input';
-
-jest.mock('flatpickr');
-jest.mock('is-touch-device');
-
-// user is in London
-// eslint-disable-next-line no-extend-native
-Date.prototype.getTimezoneOffset = () => 0;
-
-// but she wants to see time in Mardid time zone
-const timeZone = 'Europe/Madrid';
-
-const createTestProps = custom => ({
-  onChange: jest.fn(),
-  value: null,
-  placeholder: 'test',
-  intl: { formatMessage: jest.fn(message => message.id), locale: 'en' },
-  timeZone,
-  ...custom,
+import TimeInput from './time-input';
+import { render, fireEvent } from '../../../test-utils';
+
+const baseProps = { value: '', onChange: () => {} };
+
+describe('TimeInput.to24h', () => {
+  describe('when called with empty value', () => {
+    it('should return an empty string', () => {
+      expect(TimeInput.to24h('')).toEqual('');
+    });
+  });
+  describe('when called with invalid time', () => {
+    it('should return an empty string', () => {
+      expect(TimeInput.to24h('65:10')).toEqual('');
+    });
+  });
+  describe('when called with valid time', () => {
+    it('should return the time in 24h format', () => {
+      expect(TimeInput.to24h('15:10')).toEqual('15:10');
+      expect(TimeInput.to24h('15:2')).toEqual('15:02');
+      expect(TimeInput.to24h('04')).toEqual('04:00');
+      expect(TimeInput.to24h('3 AM')).toEqual('03:00');
+      expect(TimeInput.to24h('3 PM')).toEqual('15:00');
+      expect(TimeInput.to24h('3:15 AM')).toEqual('03:15');
+      expect(TimeInput.to24h('3:5 AM')).toEqual('03:05');
+      expect(TimeInput.to24h('0:00')).toEqual('00:00');
+      expect(TimeInput.to24h('10:02:03')).toEqual('10:02:03');
+      expect(TimeInput.to24h('10:2:3')).toEqual('10:02:03');
+      expect(TimeInput.to24h('10:2:3.456')).toEqual('10:02:03.456');
+      expect(TimeInput.to24h('10:2:3.5')).toEqual('10:02:03.500');
+      expect(TimeInput.to24h('10:3.5')).toEqual('');
+      expect(TimeInput.to24h('1300:00.000')).toEqual('');
+      expect(TimeInput.to24h('1300')).toEqual('');
+      expect(TimeInput.to24h('300')).toEqual('');
+      expect(TimeInput.to24h('13:00.000')).toEqual('');
+      expect(TimeInput.to24h('15:09.300')).toEqual('');
+      expect(TimeInput.to24h('10:3.5')).toEqual('');
+    });
+  });
```

Comment:
Nice unit tests! Just one thought: should we group them somehow if possible? 

```txt
describe: with padded time
describe: with unpadded time
describe: with second accuracy
describe: with high precision time
```

Just a thought as it may document the "spec" for the component a bit better.